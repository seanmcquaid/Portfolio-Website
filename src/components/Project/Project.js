import React from "react";
import styled from "styled-components";
import { ParagraphText, ExternalLinkButton } from "components/components";
import PropTypes from "prop-types";

const Project = ({title, image, description, repoLink}) => (
    <ProjectContainer>
        <ProjectTitle>{title}</ProjectTitle>
        <ProjectImage src={image} alt={title}/>
        <ParagraphText>{description}</ParagraphText>
        <ButtonsContainer>
            <ExternalLinkButton href={repoLink} target="blank">Github</ExternalLinkButton>
        </ButtonsContainer>
    </ProjectContainer>
);

const ProjectContainer = styled.div`
    border : 2px solid #00000096;
    margin : 1rem;
    display : flex;
    flex-direction : column;
    justify-content : center;
    align-items : center;
    width : 100%;
    max-width : 400px;
    height : 600px;
`;

const ProjectTitle = styled.h4`
    font-size : 1.25rem;
`;

const ProjectImage = styled.img`
    border-radius : 5px;
    box-shadow : 0px 0px 20px rgba(0,0,0,.4);
    margin : 1rem;
    width : 100%;
    max-width : 300px;
    height : 200px;
`;

const ButtonsContainer = styled.div`
    margin : 1rem;
    display : flex;
    flex-direction : row;
    justify-content : center;
    align-items : center;
    width : 100%;
    max-width : 100px;
`;

Project.propTypes = {
    title : PropTypes.string.isRequired, 
    image : PropTypes.string.isRequired, 
    description : PropTypes.string.isRequired, 
    repoLink : PropTypes.string.isRequired,
};

Project.defaultProps = {
    title : "Project Here", 
    image : "Placeholder Image", 
    description : "Placeholder description", 
    repoLink : "www.google.com",
};


export default Project;