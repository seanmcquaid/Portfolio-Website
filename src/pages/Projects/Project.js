import React from 'react';
import styled from 'styled-components';
import { ParagraphText, ExternalLinkButton } from 'components/components';
import PropTypes from 'prop-types';

const Project = ({ title, image, description, repoLink, projectId }) => (
  <ProjectContainer
    id={'projectInfo' + projectId}
    data-testid={'projectInfo' + projectId}
  >
    <ProjectTitle
      id={'projectTitle' + projectId}
      data-testid={'projectTitle' + projectId}
    >
      {title}
    </ProjectTitle>
    <ProjectImage
      src={image}
      alt={title}
      id={'projectImage' + projectId}
      data-testid={'projectImage' + projectId}
    />
    <ParagraphText
      id={'projectDescription' + projectId}
      data-testid={'projectDescription' + projectId}
    >
      {description}
    </ParagraphText>
    <ExternalLinkButton
      href={repoLink}
      target="blank"
      id={'projectRepoButton' + projectId}
      data-testid={'projectRepoButton' + projectId}
      title={`${title} GitHub Repo Link`}
    >
      GitHub Repo
    </ExternalLinkButton>
  </ProjectContainer>
);

const ProjectContainer = styled.li`
  border: 2px solid #00000096;
  margin: 1rem;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 26rem;
  height: 28rem;
  @media (max-width: 975px) {
    margin: 1rem 0;
    padding: 1rem 0;
  }
`;

const ProjectTitle = styled.label`
  font-size: 1.25rem;
  color: white;
  padding: 0.5rem;
`;

const ProjectImage = styled.img`
  border-radius: 5px;
  box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.4);
  max-width: 300px;
`;

Project.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  repoLink: PropTypes.string.isRequired,
  projectId: PropTypes.number.isRequired,
};

Project.defaultProps = {
  title: 'Project Here',
  image: 'Placeholder Image',
  description: 'Placeholder description',
  repoLink: 'www.google.com',
};

export default Project;
