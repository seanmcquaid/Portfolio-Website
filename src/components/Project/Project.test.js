import React from 'react';
import { render } from '@testing-library/react';
import Project from './Project';

describe('<Project/>', () => {
  it('Snapshot matches', () => {
    const props = {
      title: 'Portfolio',
      image: 'https://via.placeholder.com/150',
      description: 'Portfolio website',
      repoLink: 'www.github.com/seanmcquaid',
      projectId: 1,
    };
    const project = render(<Project {...props} />);
    expect(project).toMatchSnapshot();
  });
});
