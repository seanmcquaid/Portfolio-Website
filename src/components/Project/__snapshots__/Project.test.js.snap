// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project/> Snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzoiQi kWYkrd"
        data-testid="projectInfo1"
        id="projectInfo1"
      >
        <h4
          class="sc-fzqARJ eurAtz"
          data-testid="projectTitle1"
          id="projectTitle1"
        >
          Portfolio
        </h4>
        <img
          alt="Portfolio"
          class="sc-fzqNqU hJCjwJ"
          data-testid="projectImage1"
          id="projectImage1"
          src="https://via.placeholder.com/150"
        />
        <p
          class="sc-fzpjYC eJEkNF"
          data-testid="projectDescription1"
          id="projectDescription1"
        >
          Portfolio website
        </p>
        <div
          class="sc-fzoyTs dqxPnh"
        >
          <a
            class="sc-fznWqX kKWfFq"
            data-testid="projectRepoButton1"
            href="www.github.com/seanmcquaid"
            id="projectRepoButton1"
            target="blank"
          >
            GitHub Repo
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzoiQi kWYkrd"
      data-testid="projectInfo1"
      id="projectInfo1"
    >
      <h4
        class="sc-fzqARJ eurAtz"
        data-testid="projectTitle1"
        id="projectTitle1"
      >
        Portfolio
      </h4>
      <img
        alt="Portfolio"
        class="sc-fzqNqU hJCjwJ"
        data-testid="projectImage1"
        id="projectImage1"
        src="https://via.placeholder.com/150"
      />
      <p
        class="sc-fzpjYC eJEkNF"
        data-testid="projectDescription1"
        id="projectDescription1"
      >
        Portfolio website
      </p>
      <div
        class="sc-fzoyTs dqxPnh"
      >
        <a
          class="sc-fznWqX kKWfFq"
          data-testid="projectRepoButton1"
          href="www.github.com/seanmcquaid"
          id="projectRepoButton1"
          target="blank"
        >
          GitHub Repo
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
